core/logger/logger.go

```diff
@@ -194,12 +195,11 @@ func NewLogger() (Logger, func() error) {
 	if invalid != "" {
 		parseErrs = append(parseErrs, invalid)
 	}
-	if fileMaxSize == 0 || invalid != "" {
-		c.FileMaxSizeMB = 5120 // our default (env var) is 5120mb
-		parseErrs = append(parseErrs, "LogFileMaxSize will default to 5120mb")
+	if fileMaxSize <= 0 {
+		c.FileMaxSizeMB = 0 // disabled
```

Comment:
Zero is valid and means disabled.
Separately, if `invalid != ""`, `fileMaxSize` will already have the default filled in automatically, so no need to handle that case.